### PR TITLE
Update onnx to work with numpy>=2.0

### PR DESCRIPTION
**Pull request by xadupre**
### Description
onnx compiled with numpy 1.x does not work with numpy 2.0. onnx compiled with numpy 2.0 works numpy 1.x.

Fixes https://github.com/onnx/onnx/issues/6136 Fixes https://github.com/onnx/onnx/issues/6197

### CODE REVIEW

**Review by justinchuby — APPROVED** [review]
Thanks! Do we specify numpy>2.0 explicitly somewhere in the release pipeline too? If not should we do it?

**Comment by justinchuby** [comment]
Fixes https://github.com/onnx/onnx/issues/6136 ?

**Comment by xadupre** [comment]
> Fixes #6136 ?

It should.

File: onnx/test/shape_inference_test.py [hunk 1]
```diff
@@ -10172,6 +10172,66 @@ def test_check_type_when_schema_has_empty_io(self):
             op_schema.name, op_schema.since_version, op_schema.domain
         )
 
+    def test_issue_layer_normalization_6187(self):
```
Comment (justinchuby) [inline onnx/test/shape_inference_test.py]: Prefer a descriptive test name.

**Review by liqunfu — APPROVED** [review]

**Review by gramalingam — APPROVED** [review]

**Pull request merged**